### PR TITLE
Add user friend request

### PR DESCRIPTION
**Pull request by brandonbothell**

### CODE REVIEW

File: src/entity/User/User.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
-import { Entity, PrimaryColumn, Column, OneToOne } from 'typeorm'
+import { Entity, PrimaryColumn, Column, OneToOne, OneToMany } from 'typeorm'
 import { IsFQDN } from 'class-validator'
 import { UserSettings, UserVerification, UserLevel, UserBalance, UserProfile, UserReputation } from '.'
+import { UserFriendRequest } from './UserFriendRequest'
```
Comment (tgoins) [inline src/entity/User/User.ts]: Add this table to the index.ts file so you can combine the import into line 3.

File: src/entity/User/User.ts [hunk 2]
```diff
@@ -46,6 +47,12 @@ export class User {
   @OneToOne(type => UserReputation, userReputation => userReputation.user)
   reputation: UserReputation
 
+  @OneToMany(type => UserFriendRequest, userFriendRequest => userFriendRequest.user)
+  outgoingFriendRequests: UserFriendRequest
```
Comment (tgoins) [inline src/entity/User/User.ts]: Should be `UserFriendRequest[]`

File: src/entity/User/User.ts [hunk 3]
```diff
@@ -46,6 +47,12 @@ export class User {
...
+
+  @OneToMany(type => UserFriendRequest, userFriendRequest => userFriendRequest.receiver)
+  incomingFriendRequests: UserFriendRequest
```
Comment (tgoins) [inline src/entity/User/User.ts]: Should be `UserFriendRequest[]`

File: src/entity/User/UserFriendRequest.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+} from 'typeorm'
+import { User } from '.'
+
+@Entity()
+@Index(['user', 'receiver'], { unique: true })
+export class UserFriendRequest {
+  @PrimaryGeneratedColumn() id: number
+
+  @ManyToOne(type => User, user => user.outgoingFriendRequests)
+  user: User
```
Comment (tgoins) [inline src/entity/User/UserFriendRequest.ts]: add `Index()` above this.

File: src/entity/User/UserFriendRequest.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+  @ManyToOne(type => User, user => user.incomingFriendRequests)
+  receiver: User
```
Comment (tgoins) [inline src/entity/User/UserFriendRequest.ts]: add `Index()` above this.

**Review by tgoins — CHANGES_REQUESTED** [review]
I like it. Theres a few small things though.

**Comment by brandonbothell** [comment]
@tannerz28 done

**Review by tgoins — DISMISSED** [review]

**Review by tgoins — APPROVED** [review]

**Pull request merged**